kafka/conn.py

```diff
@@ -256,6 +256,7 @@ def __init__(self, host, port, afi, **configs):
         for key in self.config:
             if key in configs:
                 self.config[key] = configs[key]
+        self.config['host'] = host
```

Comment:
Since this is only used in sasl mechanisms, I think it would be better to limit the scope. How about adding `host=self.host` on L316 instead?